lxd/metrics/api_rates.go

```diff
@@ -0,0 +1,93 @@
+package metrics
+
+import (
+	"net/url"
+	"sync/atomic"
+
+	"github.com/canonical/lxd/shared/entity"
+)
+
+// RequestStatus represents the a completed request status category
+type requestStatus string
+
+const (
+	serverError requestStatus = "server_error"
+	clientError requestStatus = "client_error"
+	success     requestStatus = "succeeded"
+)
+
+// PossibleRequestStatus is a slice that includes all the possible request status categories
+var PossibleRequestStatus = []requestStatus{
+	serverError,
+	clientError,
+	success,
+}
+
+func resolveStatusCode(statusCode int, asyncOperation bool) requestStatus {
+	if asyncOperation {
```

Comment:
im not yet following why we are differentiating asyncOperations here

Reply:
I will add some comments here to make this clearer, `asyncOperation` indicates wether the status is from the spawned operation instead of the request itself. When that is the case, a 400 status code is a failure (see [this](https://documentation.ubuntu.com/lxd/en/latest/rest-api/#list-of-current-status-codes)) and the rest are considered as a success for the purpose of these metrics.